librz/arch/p/analysis/analysis_hppa_cs.c

```diff
@@ -0,0 +1,493 @@
+// SPDX-FileCopyrightText: 2024 Anton Kochkov <anton.kochkov@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_types.h>
+#include <rz_lib.h>
+#include <rz_asm.h>
+#include <capstone/capstone.h>
+
+#include <hppa/hppa.inc>
+
+static char *hppa_reg_profile(RzAnalysis *analysis) {
+	if (analysis->bits == 64) {
+		const char *p =
+			"=PC	pc\n"
+			"=SP	r30\n"
+			"=A0	r26\n"
+			"=A1	r25\n"
+			"=A2	r24\n"
+			"=A3	r23\n"
+			"gpr	r0	.64	0	0\n"
+			"gpr	r1	.64	8	0\n"
+			"gpr	r2	.64	16	0\n"
+			"gpr	r3	.64	24	0\n"
+			"gpr	r4	.64	32	0\n"
+			"gpr	r5	.64	40	0\n"
+			"gpr	r6	.64	48	0\n"
+			"gpr	r7	.64	56	0\n"
+			"gpr	r8	.64	64	0\n"
+			"gpr	r9	.64	72	0\n"
+			"gpr	r10	.64	80	0\n"
+			"gpr	r11	.64	88	0\n"
+			"gpr	r12	.64	96	0\n"
+			"gpr	r13	.64	104	0\n"
+			"gpr	r14	.64	112	0\n"
+			"gpr	r15	.64	120	0\n"
+			"gpr	r16	.64	128	0\n"
+			"gpr	r17	.64	136	0\n"
+			"gpr	r18	.64	144	0\n"
+			"gpr	r19	.64	152	0\n"
+			"gpr	r20	.64	160	0\n"
+			"gpr	r21	.64	168	0\n"
+			"gpr	r22	.64	176	0\n"
+			"gpr	r23	.64	184	0\n"
+			"gpr	r24	.64	192	0\n"
+			"gpr	r25	.64	200	0\n"
+			"gpr	r26	.64	208	0\n"
+			"gpr	r27	.64	216	0\n"
+			"gpr	r28	.64	224	0\n"
+			"gpr	r29	.64	232	0\n"
+			"gpr	r30	.64	240	0\n"
+			"gpr	r31	.64	248	0\n"
+			"ctr	sr0	.64	256	0\n"
+			"ctr	sr1	.64	264	0\n"
+			"ctr	sr2	.64	272	0\n"
+			"ctr	sr3	.64	280	0\n"
+			"ctr	sr4	.64	288	0\n"
+			"ctr	sr5	.64	296	0\n"
+			"ctr	sr6	.64	304	0\n"
+			"ctr	sr7	.64	312	0\n"
+			"flg	psw	.64	320	0\n";
+		return strdup(p);
+	} else {
+		const char *p =
+			"=PC	pc\n"
+			"=SP	r30\n"
+			"=A0	r26\n"
+			"=A1	r25\n"
+			"=A2	r24\n"
+			"=A3	r23\n"
+			"gpr	r0	.32	0	0\n"
+			"gpr	r1	.32	4	0\n"
+			"gpr	r2	.32	8	0\n"
+			"gpr	r3	.32	12	0\n"
+			"gpr	r4	.32	16	0\n"
+			"gpr	r5	.32	20	0\n"
+			"gpr	r6	.32	24	0\n"
+			"gpr	r7	.32	28	0\n"
+			"gpr	r8	.32	32	0\n"
+			"gpr	r9	.32	36	0\n"
+			"gpr	r10	.32	40	0\n"
+			"gpr	r11	.32	44	0\n"
+			"gpr	r12	.32	48	0\n"
+			"gpr	r13	.32	52	0\n"
+			"gpr	r14	.32	56	0\n"
+			"gpr	r15	.32	60	0\n"
+			"gpr	r16	.32	64	0\n"
+			"gpr	r17	.32	68	0\n"
+			"gpr	r18	.32	72	0\n"
+			"gpr	r19	.32	76	0\n"
+			"gpr	r20	.32	80	0\n"
+			"gpr	r21	.32	84	0\n"
+			"gpr	r22	.32	88	0\n"
+			"gpr	r23	.32	92	0\n"
+			"gpr	r24	.32	96	0\n"
+			"gpr	r25	.32	100	0\n"
+			"gpr	r26	.32	104	0\n"
+			"gpr	r27	.32	108	0\n"
+			"gpr	r28	.32	112	0\n"
+			"gpr	r29	.32	116	0\n"
+			"gpr	r30	.32	120	0\n"
+			"gpr	r31	.32	124	0\n"
+			"ctr	sr0	.32	128	0\n"
+			"ctr	sr1	.32	132	0\n"
+			"ctr	sr2	.32	136	0\n"
+			"ctr	sr3	.32	140	0\n"
+			"ctr	sr4	.32	144	0\n"
+			"ctr	sr5	.32	148	0\n"
+			"ctr	sr6	.32	152	0\n"
+			"ctr	sr7	.32	156	0\n"
+			"flg	psw	.32	160	0\n";
+		return strdup(p);
+	}
+}
+
+static inline void hppa_fillval(RzReg *rz_reg, csh handle, RzAnalysisValue *av, cs_hppa_op *hop) {
```

Comment:
```suggestion
static inline void hppa_fillval(const RzReg *rz_reg, csh handle, RzAnalysisValue *av, cs_hppa_op *hop) {
```

And constify `rz_reg_get` while you are at it please.